core/storage/filestore/impl/filesystem/filesystem_file.cpp

```diff
@@ -37,40 +37,42 @@ fc::outcome::result<void> FileSystemFile::open() noexcept {
 
```

Comment:
in previous projects we used to follow this scheme:
root:
| file.hpp // interface
| impl/file_impl.hpp // implementation
| impl/file_impl.cpp //implementation

and what concerns naming:
class File {virtual some_fun() = 0;}; // interface
class FileImpl: public File { some_fun() override {...} }; // implementation

Reply:
Well, it is a specific implementation on boost::filesystem, there are more possible implementations like C files, iostream and etc. I want to reflect the specific implementation in naming.